python/lsst/sims/photUtils/BandpassDict.py

```diff
@@ -0,0 +1,331 @@
+import copy
+import numpy
+from collections import OrderedDict
+from lsst.sims.photUtils import Bandpass, Sed
+
+__all__ = ["BandpassDict"]
+
+class BandpassDict(object):
+    """
+    This class will wrap an OrderedDict of Bandpass instantiations.
+
+    Upon instantiation, this class's constructor will resample
+    the input Bandpasses to be on the same wavelength grid (defined
+    by the first input Bandpass).  The constructor will then calculate
+    the 2-D phiArray for quick calculation of magnitudes in all
+    Bandpasses simultaneously (see the member method calcMagListFromSed).
+    """
+
+    def __init__(self, bandpassList, bandpassNameList):
+        """
+        @param [in] bandpassList is a list of Bandpass instantiations
+
+        @param [in] bandpassNameList is a list of tags to be associated
+        with those Bandpasses
+        """
+        self._bandpassDict = OrderedDict()
+        for bandpassName, bandpass in zip(bandpassNameList, bandpassList):
+
+            if bandpassName in self._bandpassDict:
+                raise RuntimeError("The bandpass %s occurs twice in your input " % bandpassName \
+                                   + "to BandpassDict")
+
+            self._bandpassDict[bandpassName] = copy.deepcopy(bandpass)
+
+        dummySed = Sed()
+        self._phiArray, self._wavelenStep = dummySed.setupPhiArray(self._bandpassDict.values())
+        self._wavelen_match = self._bandpassDict.values()[0].wavelen
+        self._nBandpasses = len(self._bandpassDict)
+
+
+    def __getitem__(self, bandpass):
+        return self._bandpassDict[bandpass]
+
+
+    def __len__(self):
+        return len(self._bandpassDict)
+
+
+    def __iter__(self):
+        for val in self._bandpassDict:
+            yield val
+
+
+    def values(self):
+        return self._bandpassDict.values()
+
+
+    def keys(self):
+        return self._bandpassDict.keys()
+
+
+    def calcMagListFromSed(self, sedobj, indices=None):
```

Comment:
We can drop the 'calc'.

I prefer 'From' to 'For' because it is similar to the naming convention used in sims_coordUtils methods (i.e. pixelCoordsFromRaDec) where 'From' indicates what the input should be.

We are going to have to have methods that just return numpy arrays of magnitudes so that they can be used in the catUtils getters (which expect numpy arrays).  I can add another method that converts those numpy arrays into an OrderedDict.